pallets/funding/src/tests/4_community.rs

```diff
@@ -1,1224 +1,1394 @@
+			};
+			let contribute = |inst: &mut MockInstantiator, project_id, multiplier| {
+				let jwt = get_mock_jwt(BUYER_1, InvestorType::Retail, generate_did_from_account(BUYER_1));
+				let wap = inst.get_project_details(project_id).weighted_average_price.unwrap();
+				let contributor_plmc = MockInstantiator::calculate_contributed_plmc_spent(
+					vec![(BUYER_1, 1_000 * ASSET_UNIT, Multiplier::force_new(multiplier)).into()],
+					wap,
+				);
+				let bidder_usdt = MockInstantiator::calculate_contributed_funding_asset_spent(
+					vec![(BUYER_1, 1_000 * ASSET_UNIT, Multiplier::force_new(multiplier)).into()],
+					wap,
+				);
+				let ed = MockInstantiator::get_ed();
+				inst.mint_plmc_to(vec![(BUYER_1, ed).into()]);
+				inst.mint_plmc_to(contributor_plmc);
+				inst.mint_foreign_asset_to(bidder_usdt);
+				inst.execute(|| {
+					Pallet::<TestRuntime>::community_contribute(
+						RuntimeOrigin::signed(BUYER_1),
+						jwt,
+						project_id,
+						1000 * ASSET_UNIT,
+						Multiplier::force_new(multiplier),
+						AcceptedFundingAsset::USDT,
+					)
+				})
+			};
+
+			let max_allowed_multipliers_map = vec![(2, 1), (4, 2), (9, 4), (24, 7), (25, 10)];
+
+			let mut previous_projects_created = 0;
+			for (projects_participated_amount, max_allowed_multiplier) in max_allowed_multipliers_map {
+				log::debug!("{projects_participated_amount:?}");
+
+				log::debug!("{max_allowed_multiplier:?}");
+
+				log::debug!("creating {} new projects", projects_participated_amount - previous_projects_created);
+
+				(previous_projects_created..projects_participated_amount - 1).for_each(|_| {
+					let project_id = create_project(&mut inst);
+					log::debug!("created");
+					assert_ok!(contribute(&mut inst, project_id, 1));
+				});
+
+				let project_id = create_project(&mut inst);
+				log::debug!("created");
+				previous_projects_created = projects_participated_amount;
+
+				// 0x multiplier should fail
+				// Professional bids: 0x multiplier should fail
+				inst.execute(|| {
+					assert_noop!(
+						Pallet::<TestRuntime>::community_contribute(
+							RuntimeOrigin::signed(BUYER_1),
+							get_mock_jwt(BUYER_1, InvestorType::Retail, generate_did_from_account(BUYER_1)),
+							project_id,
+							1000 * ASSET_UNIT,
+							Multiplier::force_new(0),
+							AcceptedFundingAsset::USDT
+						),
+						Error::<TestRuntime>::ForbiddenMultiplier
+					);
+				});
+
+				// Multipliers that should work
+				for multiplier in 1..=max_allowed_multiplier {
+					log::debug!("success? - multiplier: {}", multiplier);
+					assert_ok!(contribute(&mut inst, project_id, multiplier));
+				}
+
+				// Multipliers that should NOT work
+				for multiplier in max_allowed_multiplier + 1..=50 {
+					log::debug!("error? - multiplier: {}", multiplier);
+					assert_err!(
+						contribute(&mut inst, project_id, multiplier),
+						Error::<TestRuntime>::ForbiddenMultiplier
+					);
+				}
+			}
+		}
+
+		#[test]
+		fn did_with_losing_bid_can_contribute() {
```

Comment:
We maybe should skip the check on bidder 4 and 6 that are evaluators, as we do not have logic that would prevent an evaluator from contributing, right? Only people with failing bids? Something that you could test is: Evaluator does bid and uses eval PLMC to bid. Bid fails so eval plmc should be returned and can be used for contribution. Might also be an edge case, where Evaluation bond used for failing Bid should not be returned to user as unbonded PLMC? Currently I could evaluate and then create failing bid and get all my unslashable PLMC back before project ends.

Reply:
Fair point, was discussed with the team and a ticket added for further post-mvp discussion:
https://linear.app/polimec/issue/PLMC-527/future-discussion-evaluation-bond-early-release

Main takeaway is that early release is not a problem.

Checking that evaluator-bidders can participate if the bid failed is IMO still a good edge case even if like you mention we do the check only on the bid participation and not the evaluation.

The test for using evaluation plmc after having a failed bid will be added in `evaluation_bond_counts_towards_contribution`.